modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/DownloadSpreadsheetButton/DownloadSpreadsheetButton.js

```diff
@@ -0,0 +1,313 @@
+		Liferay.Language.get('cancel-and-leave')
+	);
+	const [
+		disableCancelAndLeaveButton,
+		setDisableCancelAndLeaveButton,
+	] = useState(false);
+
+	let defaultDelayTimeout = null;
+	let feedbackDelayTimeout = null;
+
+	const handleSuccess = () => {
+		setToastMessage((current) => ({
+			...current,
+			content: Liferay.Language.get('xls-was-successfully-generated'),
+			type: 'success',
+		}));
+
+		setFeedbackStatus((current) => ({
+			...current,
+			content: Liferay.Language.get('xls-generated'),
+			type: 'success',
+		}));
+	};
+
+	const handleError = ({abortedRequest, error}) => {
+		const type = abortedRequest ? 'warning' : 'danger';
+
+		const toastContent = abortedRequest
+			? Liferay.Language.get('xls-generation-has-been-cancelled')
+			: `${Liferay.Language.get(
+					'xls-generation-has-failed-try-again'
+			  )}. ${error}`;
+
+		const feedbackContent = abortedRequest
+			? Liferay.Language.get('xls-generation-cancelled')
+			: Liferay.Language.get('an-error-occurred');
+
+		setToastMessage((current) => ({
+			...current,
+			content: toastContent,
+			type,
+		}));
+
+		setFeedbackStatus((current) => ({
+			...current,
+			content: feedbackContent,
+			type,
+		}));
+	};
+
+	const handleUIState = () => {
+		defaultDelayTimeout = setTimeout(() => {
+			setLoading(false);
+
+			setFeedbackStatus((current) => ({
+				...current,
+				show: true,
+			}));
+
+			setToastMessage((current) => ({
+				...current,
+				show: true,
+			}));
+		}, DEFAULT_TIMEOUT_DELAY);
+
+		feedbackDelayTimeout = setTimeout(() => {
+			setFeedbackStatus(initialFeedbackState);
+		}, FEEDBACK_TIMEOUT_DELAY);
+	};
+
+	const buttonTextKey = loading
+		? Liferay.Language.get('generating-xls')
+		: feedbackStatus.show
+		? feedbackStatus.content
+		: Liferay.Language.get('export-xls');
+
+	const clayIconSymbol =
+		feedbackStatus.type === 'success'
+			? 'check-circle'
+			: feedbackStatus.type === 'danger'
+			? 'exclamation-circle'
+			: feedbackStatus.type === 'warning'
+			? 'warning'
+			: 'download';
+
+	const disabledGenerateButton =
+		loading || feedbackStatus.show || !parseInt(total, 10);
+
+	const handleClick = async () => {
+		setLoading(true);
+
+		try {
+			fetchController = new AbortController();
+			const blob = await fetchFile({
+				controller: fetchController,
+				url: fileURL,
+			});
+
+			downloadFileFromBlob(blob);
+
+			handleSuccess();
+		}
+		catch (error) {
+			const abortedRequest = error.name === 'AbortError' ? true : false;
+
+			handleError({abortedRequest, error});
+		}
+		finally {
+			handleUIState();
+		}
+	};
+
+	const handleCancelRequest = () => {
+		fetchController.abort();
+	};
+
+	const handleToastClose = () => {
+		setToastMessage(initialToastState);
+	};
+
+	const secondaryClickModalHandler = useCallback(
+		(onClose) => {
+			setCancelAndLeaveButtonText(Liferay.Language.get('cancelling'));
+			setDisableCancelAndLeaveButton(true);
+			handleCancelRequest();
+
+			setTimeout(() => {
```

Comment:
Should we clear this timeout as we do with the other ones?

Reply:
Absolutelly